registry/machine.go

```diff
@@ -43,17 +43,28 @@ func (r *EtcdRegistry) Machines() (machines []machine.MachineState, err error) {
 	}
 
 	for _, node := range resp.Node.Nodes {
-		for _, obj := range node.Nodes {
-			if !strings.HasSuffix(obj.Key, "/object") {
-				continue
-			}
+		var mach machine.MachineState
+
+		mach.ID = path.Base(node.Key)
 
-			var mach machine.MachineState
-			err = unmarshal(obj.Value, &mach)
-			if err != nil {
-				return
+		for _, obj := range node.Nodes {
+			// Load machine object
+			if strings.HasSuffix(obj.Key, "/object") {
+				err = unmarshal(obj.Value, &mach)
+				if err != nil {
+					return
+				}
+			} else if strings.HasSuffix(obj.Key, "/metadata") {
+				// Load metadata
+				mach.Metadata = make(map[string]string, len(obj.Nodes))
+				for _, mdnode := range obj.Nodes {
+					mach.Metadata[path.Base(mdnode.Key)] = mdnode.Value
+				}
 			}
+		}
 
+		// only use the machine if there was an object key
+		if len(mach.Version) > 0 {
```

Comment:
Fixed, originally I was pulling the machine id from the key which would not be overwritten if no object sub-key was found.